### ISSUE TITLE
http://mirage.io/xyz already redirects to https://mirage.io (the path is lost on redirections)

### ISSUE THREAD

**Issue by kit-ty-kate** [mirage/mirage-www#790]
Steps to reproduce:
1) Go to http://mirage.io/xzy
2) You are now on https://mirage.io instead of the expected https://mirage.io/xzy

**Comment by TheLortex** [comment]
That's because of that piece of code which doesn't look at the request's path:
https://github.com/mirage/mirage-www/blob/5c5a51ecdf8ecf6837fa10efa2b9497596603d24/mirage/unikernel_tls.ml#L54

**Comment by samoht** [comment]
This seems to be fixed. Please re-open if not.

**Comment by hannesm** [comment]
At least for me, this doesn't seem to be fixed. I browsed to http://mirage.io/xyz and ended up at https://mirage.io

**Comment by samoht** [comment]
Same here, that's very odd, both Chrome and Safari gives me a proper 404

**Comment by hannesm** [comment]
```
$ curl -v http://mirage.io/xyz
* Host mirage.io:80 was resolved.
* IPv6: (none)
* IPv4: 147.75.34.212
*   Trying 147.75.34.212:80...
* Connected to mirage.io (147.75.34.212) port 80
> GET /xyz HTTP/1.1
> Host: mirage.io
> User-Agent: curl/8.5.0
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Location: https://mirage.io
< Content-Length: 0
< 
* Connection #0 to host mirage.io left intact
```

And, as outlined in the initial comment, the "Location" should be "https://mirage.io/xyz".

**Comment by hannesm** [comment]
fixed in #824
